### PR TITLE
chore(test, pallas): fix deprecated shuffle in pallas test

### PR DESCRIPTION
**Pull request by jon-chuang**
This has been deprecated from JAX. Fix to supress warnings.

### CODE REVIEW

**Comment by jon-chuang** [comment]
Hello @sharadmv , this may be a minor PR, but could it be merged now (all tests pass)?

**Comment by sharadmv** [comment]
Yep will merge today

**Review by sharadmv — APPROVED** [review]

File: tests/pallas/pallas_test.py [hunk 1]
```diff
@@ -587,7 +587,7 @@ def test_array_reduce(self, op, dtype, axis):
       out_dtype = jnp.int32
     def make_x(key):
       if jnp.issubdtype(dtype, jnp.integer):
-        return random.shuffle(key, jnp.arange(m * n, dtype=dtype)).reshape(m, n)
+        return random.permutation(key, jnp.arange(m * n, dtype=dtype), independent=True).reshape(m, n)
```
Comment (sharadmv) [inline tests/pallas/pallas_test.py:590]: Actually, could you format this so it is <=80 characters per line? Thanks.

**Review by sharadmv — CHANGES_REQUESTED** [review]
Actually, left one formatting nit. LGTM otherwise.

File: tests/pallas/pallas_test.py (hunk 1 above)
Comment (jon-chuang) [reply tests/pallas/pallas_test.py]: Not sure why `pre-commit` doesn't catch it, but running flake8 on its own does. Seems bugged tbh.

image
image

Lots of other stuff is not caught by the pre-commit flake8 as well.

Comment (jon-chuang) [reply tests/pallas/pallas_test.py]: Anw, just FYI, not sure how to fix.

**Review by sharadmv — CHANGES_REQUESTED** [review]
One last thing: could you squash your commits? Then I will merge it.

**Review by sharadmv — APPROVED** [review]
